### ISSUE TITLE
OutOfMemoryError when creating a L&S game

### ISSUE THREAD

**Issue by Josharias** [MovingBlocks/Terasology#1829]
Its strange because none of the other gameplay selections causes this,  but when I run L&S with default settings I get this error:

```
13:13:32.570 [main] INFO  o.terasology.engine.TerasologyEngine - Shutting down Terasology...
java.lang.OutOfMemoryError
	at sun.misc.Unsafe.allocateMemory(Native Method)
	at java.nio.DirectByteBuffer.<init>(DirectByteBuffer.java:127)
	at java.nio.ByteBuffer.allocateDirect(ByteBuffer.java:311)
	at org.terasology.world.block.tiles.WorldAtlasImpl.createAtlasMipmaps(WorldAtlasImpl.java:273)
	at org.terasology.world.block.tiles.WorldAtlasImpl.buildAtlas(WorldAtlasImpl.java:212)
	at org.terasology.world.block.tiles.WorldAtlasImpl.<init>(WorldAtlasImpl.java:91)
	at org.terasology.engine.modes.loadProcesses.RegisterBlocks.step(RegisterBlocks.java:55)
	at org.terasology.engine.modes.StateLoading.update(StateLoading.java:240)
	at org.terasology.engine.TerasologyEngine.mainLoop(TerasologyEngine.java:497)
	at org.terasology.engine.TerasologyEngine.run(TerasologyEngine.java:425)
	at org.terasology.engine.Terasology.main(Terasology.java:158)
```

If I change the config setting from this ```"maxTextureAtlasResolution": 4096,``` to this ```"maxTextureAtlasResolution": 1024,``` it lets the world load and everything is fine.

I havent changed this setting for any of the other gameplay modules, and they dont break like this.  Not quite sure what makes it special.

Also of note, the log stops before writing the out of memory exception.

https://gist.github.com/Josharias/a2f6fee703e1d507bdb7

**Comment by Cervator** [comment]
L&S has a few unique textures and I think is the only one that retextures most the natural world tiles, for what it is worth.

Not that it really relates to the stacktrace above, but it is also the only user of Pathfinding and the workboard setup that I recall, other than the WildAnimals deer (which just uses Pathfinding)

**Comment by immortius** [comment]
Terasology currently retains a copy of each texture in memory, in addition to sending a copy to the video card. This allows retrieval of texture data (e.g. for sampling pixel data). However it really should be an option "retainInMemory", and default to false.

This would save a fair bit of memory. Beyond that we would need profile.

Regardless the texture atlas uses 4096x4096x4 bytes of data (64MB), plus more for mipmaps. If a system is low on memory anyway, or the game is being run with a low cap, then that could push it over the edge.

**Comment by Cervator** [comment]
I was able to run L&S today while merging stuff just fine, no out of memory issue here.

I did notice the items (the tools and two big ace double-blocks) didn't have working icons, as they couldn't resolve them from the "items" atlas. Could something like that be aggravating memory issues if there's a bug involved somehow?

Some of the remaining errors on L&S startup:

```
23:25:58.558 [main] WARN  o.t.r.assets.mesh.ObjMeshFormat - Skipping unsupported obj statement on line 3:"mtllib redpawntrowel.0.5.5.mtl"
23:25:58.564 [main] WARN  o.t.r.assets.mesh.ObjMeshFormat - Smoothing groups not supported in obj import yet
23:25:58.654 [main] WARN  o.t.r.assets.mesh.ObjMeshFormat - Skipping unsupported obj statement on line 3:"mtllib redpawnspear.0.5.5.mtl"
23:25:58.658 [main] WARN  o.t.r.assets.mesh.ObjMeshFormat - Smoothing groups not supported in obj import yet
23:25:58.764 [main] WARN  o.t.p.s.ComponentSerializer - Cannot deserialize unknown field 'name' onto 'engine:Item'
23:25:58.764 [main] ERROR o.t.p.t.e.TextureRegionTypeHandler - Failed to resolve texture region 'LightAndShadowResources:items.club'
...
3:25:58.773 [main] WARN  o.t.r.assets.mesh.ObjMeshFormat - Skipping unsupported obj statement on line 3:"mtllib redpawnhammer.0.5.5.mtl"
23:25:58.777 [main] WARN  o.t.r.assets.mesh.ObjMeshFormat - Smoothing groups not supported in obj import yet
23:25:58.789 [main] WARN  o.t.p.s.ComponentSerializer - Cannot deserialize unknown field 'name' onto 'engine:Item'
23:25:58.789 [main] ERROR o.t.p.t.e.TextureRegionTypeHandler - Failed to resolve texture region 'engine:items.walkToJob'
23:25:58.855 [main] WARN  o.t.p.s.ComponentSerializer - Cannot deserialize unknown field 'groundFriction' onto 'engine:CharacterMovement'
23:25:58.855 [main] WARN  o.t.p.s.ComponentSerializer - Cannot deserialize unknown field 'maxGroundSpeed' onto 'engine:CharacterMovement'
23:25:58.855 [main] WARN  o.t.p.s.ComponentSerializer - Cannot deserialize unknown field 'distanceBetweenSwimStrokes' onto 'engine:CharacterMovement'
23:25:58.993 [main] ERROR o.t.p.serializers.PrefabSerializer - Prefab contains unknown component 'Breather'
23:25:59.019 [main] WARN  o.t.p.s.ComponentSerializer - Cannot deserialize unknown field 'name' onto 'engine:Item'
...
23:25:59.043 [main] WARN  o.t.p.s.ComponentSerializer - Cannot deserialize unknown field 'name' onto 'engine:Item'
23:25:59.043 [main] ERROR o.t.p.t.e.TextureRegionTypeHandler - Failed to resolve texture region 'LightAndShadowResources:items.heart'
```

**Comment by Cervator** [comment]
@Josharias do you still get the memory issue? I fixed a few icons and created #1840 but have yet to run out of memory in L&S. Which could make sense if it is more system specific.

**Comment by Josharias** [comment]
Today I do not get this same out of memory error.  Ill check again tomorrow to see if it crops up again.  Could have been just how my system as behaving that day.
